tests/utils/test_cache.py

```diff
@@ -0,0 +1,160 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+from typing import Any
+from typing import Union
+
+import pytest
+
+from cachy import CacheManager
+
+from poetry.utils.cache import FileCache
+
+
+if TYPE_CHECKING:
+    from pathlib import Path
+
+    from _pytest.monkeypatch import MonkeyPatch
+    from pytest import FixtureRequest
+    from pytest_mock import MockerFixture
+
+    from tests.conftest import Config
+
+
+FILE_CACHE = Union[FileCache, CacheManager]
+
+
+@pytest.fixture
+def repository_cache_dir(monkeypatch: MonkeyPatch, config: Config) -> Path:
+    return config.repository_cache_directory
+
+
+def patch_cachy(cache: CacheManager) -> CacheManager:
+    old_put = cache.put
+    old_remember = cache.remember
+
+    def new_put(key: str, value: Any, minutes: int | None = None):
+        if minutes is not None:
+            return old_put(key, value, minutes=minutes)
+        else:
+            return cache.forever(key, value)
+
+    cache.put = new_put
+
+    def new_remember(key: str, value: Any, minutes: int | None = None):
+        if minutes is not None:
+            return old_remember(key, value, minutes=minutes)
+        else:
+            return cache.remember_forever(key, value)
+
+    cache.remember = new_remember
+    return cache
+
+
+@pytest.fixture
+def cachy_file_cache(repository_cache_dir: Path) -> CacheManager:
+    cache = CacheManager(
+        {
+            "default": "cache",
+            "serializer": "json",
+            "stores": {
+                "cache": {"driver": "file", "path": str(repository_cache_dir / "cache")}
+            },
+        }
+    )
+    return patch_cachy(cache)
+
+
+@pytest.fixture
+def file_cache(repository_cache_dir: Path) -> FileCache:
```

Comment:
Let's call this `poetry_file_cache` or `builtin_file_cache` for clarity.

Reply:
Changed it to `poetry_file_cache`, added a `remember` test as well.